activerecord/lib/active_record/database_configurations.rb

```diff
@@ -36,9 +36,11 @@ class InvalidConfigurationError < StandardError; end
     # to respond to `sharded?`. To implement this define the following in an
     # initializer:
     #
-    #   ActiveRecord::DatabaseConfigurations.register_db_config_handler do |env_name, name, url, config|
-    #     next unless config.key?(:vitess)
-    #     VitessConfig.new(env_name, name, config)
+    #   ActiveSupport.on_load(:active_record_database_configurations) do
+    #     ActiveRecord::DatabaseConfigurations.register_db_config_handler do |env_name, name, url, config|
+    #       next unless config.key?(:vitess)
+    #       VitessConfig.new(env_name, name, config)
+    #     end
     #   end
```

Comment:
Could this code example demonstrate the fact that the hook executes within the context of the `ActiveRecord::DatabaseConfigurations` class?

```diff
    #   ActiveSupport.on_load(:active_record_database_configurations) do
-   #     ActiveRecord::DatabaseConfigurations.register_db_config_handler do |env_name, name, url, config|
+   #     register_db_config_handler do |env_name, name, url, config|
    #       next unless config.key?(:vitess)
    #       VitessConfig.new(env_name, name, config)
    #     end
```

Reply:
Ooh, I do like that better, if I'm in this code again I will update the comment.